src/path.cpp

```diff
@@ -275,7 +275,7 @@ static void maybe_issue_path_warning(const wcstring &which_dir, const wcstring &
         debug(0, _(L"The error was '%s'."), strerror(saved_errno));
         debug(0, _(L"Please set $%ls to a directory where you have write access."), env_var);
     }
-    write(STDERR_FILENO, "\n", 1);
+    write_ignore(STDERR_FILENO, "\n", 1);
```

Comment:
In practice there isn't anything fish can do that would be sensible if a write to stdout or stderr fails. The most common source of an error is the tty being closed, a full filesystem (errno ENOSPC) or the read side of a pipe being closed (errno EPIPE). Often the simplest way to deal with these failures is to ignore them and simply let the script run to completion. The alternative is to exit early when the first error occurs.